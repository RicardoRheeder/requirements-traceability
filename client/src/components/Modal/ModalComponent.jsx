import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Modal from 'react-modal'
import { setModalObject } from '../../redux/stores/common/actions'

import { Modal_AddDocument, Modal_RemoveDocument } from './'

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
}

export default function ModalComponent() {
  const modalObject = useSelector((state) => state.common.modalObject, [])
  const dispatch = useDispatch()

  const closeModal = () => {
    dispatch(setModalObject({ visible: false, mode: 0 }))
  }

  return (
    <div className="modal-root">
      {modalObject.visible ? (
        <Modal
          isOpen={modalObject.visible}
          onRequestClose={closeModal}
          style={customStyles}
          contentLabel="Example Modal"
        >
          {modalObject.mode == 0 ? (
            <Modal_AddDocument />
          ) : modalObject.mode == 1 ? (
            <Modal_RemoveDocument />
          ) : (
            <></>
          )}
          <button onClick={closeModal}>close</button>
        </Modal>
      ) : (
        <></>
      )}
    </div>
  )
}
