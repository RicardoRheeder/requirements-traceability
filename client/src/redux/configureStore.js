import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
