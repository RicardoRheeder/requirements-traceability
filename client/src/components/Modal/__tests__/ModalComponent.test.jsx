import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import ModalComponent from "../ModalComponent"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("modal component base", () => {
    let store
    beforeEach(() => {
        store = mockStore({common:{
            modalObject: { visible: true}
        }});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><ModalComponent/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><ModalComponent/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});