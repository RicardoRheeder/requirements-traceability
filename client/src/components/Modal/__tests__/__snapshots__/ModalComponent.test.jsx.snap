// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal component base renders 1`] = `ShallowWrapper {}`;

exports[`modal component base renders with children 1`] = `ReactWrapper {}`;
